libraries/AC_WPNav/AC_Circle.cpp

```diff
@@ -152,6 +153,23 @@ bool AC_Circle::is_active() const
     return (AP_HAL::millis() - _last_update_ms < 200);
 }
 
+// get rate in degrees per second. Rate will be +ve for clockwise and -ve for counter-clockwise
+float AC_Circle::get_rate_with_direction() const
+{
+    switch (_direction)
+    {
+    case TurnDirection::CIRCLE_CW:
+        return fabsf(_rate);
```

Comment:
why not just encode the direction in _rate?

Reply:
Yeah, I was thinking about it; it would make it much simpler. But that would mean if we go to circle mode after auto, we might have an inverted turn. We could save the original "_rate" and return back to default on exiting auto though..